include/nuttx/irq.h

```diff
@@ -70,6 +70,18 @@
 
 #endif /* __ASSEMBLY__ */
 
+#ifdef CONFIG_SMP
+#  define cpu_irqlock_clear() \
+  do \
+    { \
+      g_cpu_irqset = 0; \
+      SP_DMB(); \
+      g_cpu_irqlock = SP_UNLOCKED; \
+      SP_DSB(); \
+    } \
+  while (0)
```

Comment:
nit: maybe it can be
```
#  define cpu_irqlock_clear(cpu) \
  do \
    { \
      g_cpu_irqset &= ~(1 << cpu); \
      SP_DMB(); \
      g_cpu_irqlock = SP_UNLOCKED; \
      SP_DSB(); \
    } \
  while (0)
```
similar with `cpu_irqlock_set(cpu)`
There is a benefit, it can be used to trace is there different cpus enter critical section simultaneously.
WDYT?

Reply:
"g_cpu_irqset = 0" can save at least one CPU instruction compared to "g_cpu_irqset &= ~(1 << cpu);"
If adding "DEBUGASSERT(g_cpu_irqset == 0);" in "cpu_irqlock_set(cpu)" can also achieve the effect you want?
for example:
#  define cpu_irqlock_set(cpu) \
  do \
    { \
      DEBUGASSERT(g_cpu_irqset == 0); \
      g_cpu_irqset |= (1 << cpu); \
    } \
  while (0)

Reply:
Got it, thank for your explaination. This idea is very great. Thanks.